src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/HttpClient/SendAsync.cs

```diff
@@ -10,38 +10,29 @@
 using NewRelic.Agent.Api;
 using NewRelic.Agent.Api.Experimental;
 using NewRelic.Agent.Extensions.Providers.Wrapper;
-using NewRelic.SystemExtensions;
 
 namespace NewRelic.Providers.Wrapper.HttpClient
 {
-    public class SendAsync : IWrapper
+    public class SendAsyncFactory : MethodWrapper<System.Net.Http.HttpClient>
     {
-        public const string InstrumentedTypeName = "System.Net.Http.HttpClient";
-
-        public bool IsTransactionRequired => true;
-
-        public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
+        public SendAsyncFactory(IAgent agent, ITransaction transaction, InstrumentedMethodCall instrumentedMethodCall, System.Net.Http.HttpClient invocationTarget)
+            : base(agent, transaction, instrumentedMethodCall, invocationTarget)
         {
-            var method = methodInfo.Method;
-            if (method.MatchesAny(assemblyName: "System.Net.Http", typeName: InstrumentedTypeName, methodName: "SendAsync"))
-            {
-                return TaskFriendlySyncContextValidator.CanWrapAsyncMethod("System.Net.Http", "System.Net.Http.HttpClient", method.MethodName);
-            }
-            else
-            {
-                return new CanWrapResponse(false);
-            }
         }
 
-        public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction)
+        [MethodWrapper(UsesContinuation = true)]
```

Comment:
This is an example usage of the `MethodWrapper` attribute for a wrapper for the `HttpClient.SendAsync` method. The "wrapper" method has the same signature as `SendAsync`.